app/services/request_password_reset.rb

```diff
@@ -2,6 +2,7 @@
   def perform
     if user_should_receive_registration_email?
       form = RegisterUserEmailForm.new
+      form.for_password_reset = true
```

Comment:
I think this is a weird way to set a property after initializing. What if we updated the constructor to take keyword arguments? Then we could preserve most of the old behavior with minimal changes to calling sites?

```ruby
def initialize(for_password_reset: false, recaptcha_results: [true, {}]))
```
then this becomes:
```ruby
form = RegisterUserEmailForm.new(for_password_reset: true)
```
```

Reply:
This means I'd have to go anywhere `recaptcha_results ` is currently set in the constructor and change the signature to `new(recaptcha_results: <value>`, right?  I was admittedly trying to avoid doing that by using the post-init attribute update.  But, I totally agree that keyword signatures are awesome, and I'll look into it right now.

Reply:
Luckily `RegisterUserEmailForm.new` is only called with recaptcha results in one, place: `registrations_controller.rb`